### PR TITLE
txn: some refinements to reduce duplicate code

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

## What have you changed? (mandatory)
some refinements

## What are the type of the changes? (mandatory)
- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)
all tests

## Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/pull/4698

### CODE REVIEW

**Comment by youjiali1995** [comment]
@coocood @AndreMouche @zhangjinpeng1987 PTAL

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -212,14 +250,7 @@ impl<S: Snapshot> MvccTxn<S> {
...
             if should_not_exist {
-                if write.write_type == WriteType::Put
-                    || (write.write_type != WriteType::Delete
-                        && self.key_exist(&key, write.start_ts - 1)?)
-                {
-                    return Err(Error::AlreadyExist {
-                        key: key.into_raw()?,
-                    });
-                }
+                self.check_data_constraint(&write, &key)?;
```
Comment (coocood) [inline src/storage/mvcc/txn.rs]: I think it's better to put `should_not_exist` as an argument to `check_data_constraint`.

Comment (youjiali1995) [reply src/storage/mvcc/txn.rs]: fixed

**Comment by coocood** [comment]
LGTM

**Comment by youjiali1995** [comment]
@zhangjinpeng1987 @AndreMouche @MyonKeminta PTAL

File: src/storage/mvcc/txn.rs [hunk 2]
```diff
@@ -188,15 +244,20 @@ impl<S: Snapshot> MvccTxn<S> {
...
+                if commit_ts == self.start_ts {
+                    assert!(write.write_type == WriteType::Rollback);
+                    return Err(Error::PessimisticLockRollbacked {
+                        start_ts: self.start_ts,
+                        key: key.into_raw()?,
+                    });
+                } else {
+                    // If the `commit_ts` is not equal to `start_ts`,
+                    // the transaction has been commited.
+                    return Err(Error::Committed { commit_ts });
```
[the remark this answers is not in the document]
Comment (youjiali1995) [reply src/storage/mvcc/txn.rs]: Yes, it is impossible.

**Comment by youjiali1995** [comment]
I have removed the useless check. Please take a look again. @zhangjinpeng1987 @MyonKeminta

**Pull request merged**